File: ui/hb-app/src/app/transactions/models/OneTimeTransactionDto.ts
import {BankAccountDto} from "../../component-bank-account/models/bank-account-dto";
import {CostDirection} from "./CostDirection";

export class OneTimeTransactionDto {
  private _bankAccount: BankAccountDto;

  constructor();
  constructor(
    public id?: string,
    public name?: string,
    public value?: number,
    public costDirection?: CostDirection,
    public payDate?: Date,
    public bankAccountId?: number
  ) {
  }

  get bankAccount(): BankAccountDto {
    return this._bankAccount;
  }

  set bankAccount(value: BankAccountDto) {
    this._bankAccount = value;
  }

  get(){
    this.payDate = new Date(this.payDate);
    return this;
  }
}

File: src/main/java/com/urban/gosia/transaction/models/dto/OneTimeTransactionDto.java
package com.urban.gosia.transaction.models.dto;

import com.urban.gosia.transaction.models.OneTimeTransaction;
import com.urban.gosia.transaction.models.Transaction;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Date;
import java.util.UUID;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
public class OneTimeTransactionDto {
    private UUID id;
    private String name;
    private BigDecimal value;
    private Transaction.CostDirection costDirection;
    private Date payDate;
    private Integer bankAccountId;

    public static OneTimeTransactionDto convert(OneTimeTransaction t) {
        return OneTimeTransactionDto.builder()
                .id(t.getId())
                .name(t.getName())
                .value(t.getValue())
                .costDirection(t.getCostDirection())
                .payDate(t.getPayDate())
                .bankAccountId(t.getBankAccount() != null ? t.getBankAccount().getId() : null)
                .build();
    }
}
